src/tiledb/cloud/utilities/_common.py

```diff
@@ -347,3 +382,94 @@ def wrapper(*args, **kwargs) -> Dict[str, object]:
 
     return wrapper
+
+
+def find(
+    uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    include: Optional[Union[str, Callable]] = None,
+    exclude: Optional[Union[str, Callable]] = None,
+    max_count: Optional[int] = None,
+) -> Iterator[str]:
+    """Searches a path for files matching the include/exclude pattern using VFS.
+
+    :param uri: Input path to search
+    :param config: Optional dict configuration to pass on tiledb.VFS
+    :param include: Optional include pattern string
+    :param exclude: Optional exclude pattern string
+    :param max_count: Optional stop point when searching for files
+    :param count: Current number of files found
+    """
+
+    with tiledb.scope_ctx(config):
+        vfs = tiledb.VFS(config=config, ctx=tiledb.Ctx(config))
+        listing = vfs.ls(uri)
+        current_count = 0
+
+        def list_files(listing):
+            for f in listing:
+                # Avoid infinite recursion
+                if f == uri:
+                    continue
+
+                if vfs.is_dir(f):
+                    yield list_files(
+                        f,
+                        include=include,
+                        exclude=exclude,
+                    )
+                else:
+                    # Skip files that do not match the include pattern or match
+                    # the exclude pattern.
+                    if callable(include):
+                        if not include(f):
+                            continue
+                    else:
+                        if include and not fnmatch(f, include):
+                            continue
+
+                    if callable(exclude):
+                        if exclude(f):
+                            continue
+                    else:
+                        if exclude and fnmatch(f, exclude):
+                            continue
+                    yield f
+
+        for f in list_files(listing):
+            current_count += 1
+            if max_count and current_count == max_count:
+                return
+            else:
+                yield f
+
+
+T = TypeVar("T")
+
+
+def batch(items: Sequence[T], chunk_size: int) -> Iterator[Sequence[T]]:
+    """Batches a sequence of objects and returns an iterator where
+    each return sequence is of length chunk_size.
+
+    :param items: Sequence to split into batches
+    :param chunk_size: Size of chunks of the sequence to return
+    """
+    # Iterator for providing batches of chunks
+    length = len(items)
+    for ndx in range(0, length, chunk_size):
+        yield items[ndx : min(ndx + chunk_size, length)]
+
+
+def serialize_filter(filter) -> dict:
+    """Serialize TileDB filter.
+
+    :param filter: TileDB filter to serialize
+    :return: dict, TileDB filter attributes
+    """
+    if isinstance(filter, tiledb.Filter):
+        filter_dict = filter._attrs_()
+        filter_dict["_name"] = type(filter).__name__
```

Comment:
This is the same format we use in TileDB SOMA, except that we use the key `_type` rather than `_name`. Maybe useful to keep that consistent?

Reply:
forgot to put the link here: https://github.com/single-cell-data/TileDB-SOMA/blob/db548eb0d8ef09c941993c56b394aa9cdd6d1777/apis/python/src/tiledbsoma/options/_tiledb_create_options.py#L30-L36

Reply:
👍